### PR TITLE
Stop the cocktail shaker from getting reclaimed in the glass recycler

### LINKED ISSUE #1441: Shaker goes in glass recycler
Submitter: Keiranhalcyon
Cocktail shaker can be put in the glassware recycler, its metal it shouldnt be able to be put in the glassware recycler.

### PR DESCRIPTION
**Pull request by Carbadox**
##About the PR 
Prevents the players from being able to reclaim the cocktail shaker in a glass recycler

## Why's this needed? 
closes #1441

### CODE REVIEW

File: code/obj/machinery/glass_recycler.dm [hunk 1]
```diff
@@ -36,7 +36,11 @@
...
 			return 1
 		else if (istype(W, /obj/item/raw_material/shard) || istype(W, /obj/item/reagent_containers/glass) || istype(W, /obj/item/reagent_containers/food/drinks))
-			if (istype(W,/obj/item/reagent_containers/food/drinks/bottle))
+			if (istype(W,/obj/item/reagent_containers/food/drinks))
+				var/obj/item/reagent_containers/food/drinks/bottle/D = W
+				if (!D.can_recycle)
+					boutput(user, "<span class='alert'>[src] only accepts glass shards or glassware!</span>")
+					return
 				var/obj/item/reagent_containers/food/drinks/bottle/B = W
 				if (!B.broken) glass_amt += 1
```
Comment (Sovexe) [inline code/obj/machinery/glass_recycler.dm:45]:
```suggestion
			if (istype(W,/obj/item/reagent_containers/food/drinks))
				var/obj/item/reagent_containers/food/drinks/D = W
				if (!D.can_recycle)
					boutput(user, "<span class='alert'>[src] only accepts glass shards or glassware!</span>")
					return
				if (istype(W,/obj/item/reagent_containers/food/drinks/bottle))
					var/obj/item/reagent_containers/food/drinks/bottle/B = W
					if (!B.broken) glass_amt += 1
```

**Comment by Sovexe** [comment]
```
else if (istype(W, /obj/item/reagent_containers/food/drinks/drinkingglass))
			if (istype(W, /obj/item/reagent_containers/food/drinks/drinkingglass/pitcher))
				glass_amt += 2
			else
				glass_amt += 1
			user.visible_message("<span class='notice'>[user] inserts [W] into [src].</span>")
			user.u_equip(W)
			qdel(W)
			return 1
```

honestly this should probably be wrapped into this check too to clean up this proc, but it shouldn't have any functional difference for now

**Comment by pali6** [comment]
I also like the idea oh adding `glass_amount` to `/obj/item/reagent_containers/food/drinks` but if you insist on keeping it as it is it's probably ok.

**Comment by Carbadox** [comment]
Less insisting to keep it as is and more me not being able to find the motivation to work on code stuff, as of this moment.

**Comment by ZeWaka** [comment]
fine for now, i would like the other solution next time this crops up though

**Pull request merged**